File: CassandraSparkTest/src/main/java/Interaccion.java
/**
 * Created by joseluis on 6/09/17.
 */
import java.io.Serializable;
import com.datastax.driver.core.LocalDate;
import org.apache.spark.sql.Dataset;

public class Interaccion implements Serializable{
    private String idinteraccion;
    private String idorigen;
    private String tipoorigen;//Alumno, Profesor, Material
    private String iddestino;
    private String tipodestino;//Alumno, Profesor, Material
    private String tipointeraccion;
    private String valorinteraccion;
    private String idprecedente;
    private String idcurso;
    private String fecha;
    private String plataforma;//Facebook, PAM, CREA

    public Interaccion(){

    }

    public Interaccion(String idinteraccion, String idorigen, String tipoorigen, String iddestino, String tipodestino, String tipointeraccion, String valorinteraccion, String idprecedente, String idcurso, String fecha, String plataforma) {
        this.idinteraccion = idinteraccion;
        this.idorigen = idorigen;
        this.tipoorigen = tipoorigen;
        this.iddestino = iddestino;
        this.tipodestino = tipodestino;
        this.tipointeraccion = tipointeraccion;
        this.valorinteraccion = valorinteraccion;
        this.idprecedente = idprecedente;
        this.idcurso = idcurso;
        this.fecha = fecha;
        this.plataforma = plataforma;
    }

    public String getidinteraccion() {
        return idinteraccion;
    }

    public void setidinteraccion(String idinteraccion) {
        this.idinteraccion = idinteraccion;
    }

    public String getidorigen() {
        return idorigen;
    }

    public void setidorigen(String idorigen) {
        this.idorigen = idorigen;
    }

    public String gettipoorigen() {
        return tipoorigen;
    }

    public void settipoorigen(String tipoorigen) {
        this.tipoorigen = tipoorigen;
    }

    public String getiddestino() {
        return iddestino;
    }

    public void setiddestino(String iddestino) {
        this.iddestino = iddestino;
    }

    public String gettipodestino() {
        return tipodestino;
    }

    public void settipodestino(String tipodestino) {
        this.tipodestino = tipodestino;
    }

    public String gettipointeraccion() {
        return tipointeraccion;
    }

    public void settipointeraccion(String tipointeraccion) {
        this.tipointeraccion = tipointeraccion;
    }

    public String getvalorinteraccion() {
        return valorinteraccion;
    }

    public void setvalorinteraccion(String valorinteraccion) {
        this.valorinteraccion = valorinteraccion;
    }

    public String getidprecedente() {
        return idprecedente;
    }

    public void setidprecedente(String idprecedente) {
        this.idprecedente = idprecedente;
    }

    public String getidcurso() {
        return idcurso;
    }

    public void setidcurso(String idcurso) {
        this.idcurso = idcurso;
    }

    public String getfecha() {
        return fecha;
    }

    public void setfecha(String fecha) {
        this.fecha = fecha;
    }

    public String getplataforma() {
        return plataforma;
    }

    public void setplataforma(String plataforma) {
        this.plataforma = plataforma;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Interaccion that = (Interaccion) o;

        return idinteraccion != null ? idinteraccion.equals(that.idinteraccion) : that.idinteraccion == null;
    }

    @Override
    public int hashCode() {
        return idinteraccion != null ? idinteraccion.hashCode() : 0;
    }

    @Override
    public String toString() {
        return "Interaccion{" +
                "idinteraccion='" + idinteraccion + '\'' +
                '}';
    }
}

File: CSTest/src/main/scala/model/Interaccion.scala
package model

import com.datastax.spark.connector.mapper.DefaultColumnMapper

/**
  * Created by José Manuel Jiménez Ávila on 08/08/2017.
  */

class Interaccion extends Serializable {
  //Column mappings
  var idinteraccion: String = _
  var idorigen: String = _
  var tipoorigen: String = _
  var iddestino: String = _
  var tipodestino: String = _
  var tipointeraccion: String = _
  var valorinteraccion: String = _
  var idprecedente: String = _
  var idcurso: String = _
  var fecha: String = _
  var plataforma: String = _

  def this(idinteraccion: String, idorigen: String, tipoorigen: String, iddestino: String,
           tipodestino: String, tipointeraccion: String, valorinteraccion: String,
           idprecedente: String, idcurso: String, fecha: String, plataforma: String){
    this
    this.idinteraccion = idinteraccion
    this.idorigen = idorigen
    this.tipoorigen = tipoorigen
    this.iddestino = iddestino
    this.tipodestino = tipodestino
    this.tipointeraccion = tipointeraccion
    this.valorinteraccion = valorinteraccion
    this.idprecedente = idprecedente
    this.idcurso = idcurso
    this.fecha = fecha
    this.plataforma = plataforma
  }

  /*
  //Getters
  def idinteraccion: String = _idinteraccion
  def idorigen: String = _idorigen
  def tipoorigen: String = _tipoorigen
  def iddestino: String = _iddestino
  def tipodestino: String = _tipodestino
  def tipointeraccion: String = _tipointeraccion
  def valorinteraccion: String = _valorinteraccion
  def idprecedente: String = _idprecedente
  def idcurso: String = _idcurso
  def fecha: String = _fecha
  def plataforma: String = _plataforma

  //Setters
  def idinteraccion_= (value: String):Unit = _idinteraccion = value
  def idorigen_= (value: String):Unit = _idorigen = value
  def tipoorigen_= (value: String):Unit = _tipoorigen = value
  def iddestino_= (value: String):Unit = _iddestino = value
  def tipodestino_= (value: String):Unit = _tipodestino = value
  def tipointeraccion_= (value: String):Unit = _tipointeraccion = value
  def valorinteraccion_= (value: String):Unit = _valorinteraccion = value
  def idprecedente_= (value: String):Unit = _idprecedente = value
  def idcurso_= (value: String):Unit = _idcurso = value
  def fecha_= (value: String):Unit = _fecha = value
  def plataforma_= (value: String):Unit = _plataforma = value

  */

}

object Interaccion {
  /*implicit object Mapper extends DefaultColumnMapper[Interaccion](
    Map("idinteraccion_=" -> "idinteraccion", "idorigen_=" -> "idorigen",
    "tipoorigen_=" -> "tipoorigen", "iddestino_=" -> "iddestino",
    "tipodestino_=" -> "tipodestino", "tipointeraccion_=" -> "tipointeraccion",
    "valorinteraccion_=" -> "valorinteraccion", "idpredecente_=" -> "idpredecente",
    "idcurso_=" -> "idcurso", "fecha_=" -> "fecha", "plataforma_=" -> "plataforma"))*/
  implicit object Mapper extends DefaultColumnMapper[Interaccion]
}
